crates/libs/wallet/src/lib.rs

```diff
@@ -426,13 +426,42 @@ pub async fn swap_to_have_target_amount<U: Unit>(
     Ok(new_tokens)
 }
 
-pub async fn receive_wad(
+pub async fn receive_wad<U: Unit + serde::Serialize>(
     pool: Pool<SqliteConnectionManager>,
     node_client: &mut NodeClient<Channel>,
     node_id: u32,
     unit: &str,
     compact_keyset_proofs: Vec<CompactKeysetProofs>,
+    // Optional parameters for history tracking
+    history_tracking: Option<(NodeUrl, U, Option<String>)>,
```

Comment:
We don't want it to be optional.
It should always be tracked.

Is there a reason for making it optional? Like, are there situations where it is not possible?